### PR TITLE
Adds the gaseous meteor wave

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Adds a new type of meteor wave, where the meteors are filled with random gases that are released once the meteor finish their runs.
## Why It's Good For The Game
It's a different type of atmospheric issue instead of the usually caused by atmos technicians. With the new changes to scrubber (PRd) and space heaters it's easier to fix those issues and it's important that they are caused also from sources outside the department, because in most cases you have either the fire caused by atmos traitor or the fire caused by atmos error.
## Changelog
:cl:
add: Adds the gaseous meteor wave where meteors are filled with gases that releases at the end of the meteor run
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
wont those gases just get spaced

**Comment by Ghilker** [comment]
some will, many won't

File: code/modules/meteors/meteors.dm [hunk 1]
```diff
@@ -510,5 +512,45 @@ GLOBAL_LIST_INIT(meteorsSPOOKY, list(/obj/effect/meteor/pumpkin))
...
+		/datum/gas/plasma,
+		/datum/gas/nitrous_oxide,
+		/datum/gas/healium,
+		/datum/gas/freon,
+		/datum/gas/tritium,
+		/datum/gas/miasma,
+		/datum/gas/hydrogen,
+		/datum/gas/water_vapor,
+		/datum/gas/hypernoblium
+		)
```
Comment (distributivgesetz) [inline code/modules/meteors/meteors.dm:538]: gasses that do not cause an instant bad reaction or a fire will have zero effect, you want to have a tritfire/plasmafire meteor and not a co2 meteor if you wanna actually make an impact

File: code/modules/meteors/meteors.dm [hunk 2]
```diff
@@ -26,6 +26,8 @@ GLOBAL_LIST_INIT(meteorsD, list(/obj/effect/meteor/medium=15, /obj/effect/meteor
 						  /obj/effect/meteor/banana=25, /obj/effect/meteor/meaty=10, /obj/effect/meteor/meaty/xeno=8, /obj/effect/meteor/emp = 30, \
 						  /obj/effect/meteor/cluster=20, /obj/effect/meteor/tunguska=1)) //for stray meteor event (bigger numbers for a bit finer weighting)
 
+GLOBAL_LIST_INIT(meteros_gaseous, list(/obj/effect/meteor/gaseous = 1))
```
[the remark this answers is not in the document]
Comment (Ghilker) [reply code/modules/meteors/meteors.dm]: ops lol

File: code/modules/meteors/meteors.dm (hunk 1 above)
Comment (Ghilker) [reply code/modules/meteors/meteors.dm:538]: other mixes are welcome other than the standard fires

**Comment by Serxule** [comment]
meteors can cause even MORE lag? lets go

**Comment by Pickle-Coding** [comment]
> meteors can cause even MORE lag? lets go

Atmospherics almost never goes overtime. This will cause 0 extra lag in most cases. Don't comment about performance costs if you don't know how it works.

**Comment by private-tristan** [comment]
won't these just cause the gas to spread a bit and then get spaced by their entry hole?

File: code/modules/meteors/meteors.dm (hunk 1 above)
Comment (dragomagol) [inline code/modules/meteors/meteors.dm:538]:
```suggestion
		/datum/gas/hypernoblium,
	)
```

**Comment by Ghilker** [comment]
> this kinda sounds better as a stray meteor than a whole wave tbh

Stray meteor has lower probability to release gases inside

File: code/modules/meteors/meteors.dm [hunk 3]
```diff
@@ -510,5 +512,45 @@ GLOBAL_LIST_INIT(meteorsSPOOKY, list(/obj/effect/meteor/pumpkin))
...
+		/datum/gas/freon,
+		/datum/gas/tritium,
+		/datum/gas/miasma,
+		/datum/gas/hydrogen,
+		/datum/gas/water_vapor,
+		/datum/gas/hypernoblium,
+	)
+
+	var/list/picked_gases = list()
+	for(var/_ in 1 to rand(1, 3))
```
Comment (Time-Green) [inline code/modules/meteors/meteors.dm:541]: Please just use i or something, I know you don’t use it but lowercase is just cursed

**Review by Time-Green — CHANGES_REQUESTED** [review]

File: code/modules/events/meteor_wave.dm [hunk 4]
```diff
@@ -91,3 +93,18 @@
 
 /datum/round_event/meteor_wave/meaty/announce(fake)
 	priority_announce("Meaty ores have been detected on collision course with the station.", "Oh crap, get the mop.", ANNOUNCER_METEORS)
+
+/datum/round_event_control/meteor_wave/gaseous
+	name = "Meteor Wave: Gaseous"
+	typepath =  /datum/round_event/meteor_wave/gaseous
```
Comment (Time-Green) [inline code/modules/events/meteor_wave.dm:99]: Double space?

File: code/modules/events/meteor_wave.dm [hunk 5]
```diff
@@ -44,6 +44,8 @@
 			wave_type = GLOB.meteorsC
 		if("halloween")
 			wave_type = GLOB.meteorsSPOOKY
+		if("gaseous")
+			wave_type = GLOB.meteors_gaseous
```
Comment (Time-Green) [inline code/modules/events/meteor_wave.dm:48]: Whats the point of this, looks unreachable

**Comment by Time-Green** [comment]
LIVVEEEEE DAMMIT

**Pull request closed without merging**